doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+* same zero padding along both axes (:math:`p_1 = p_2 = p`).
+
+This facilitates the analysis and the visualization, but keep in mind that the
+results outlined here also generalize to the N-D and non-square cases.
+
+Theano terminology
+==================
+
+Theano has its own terminology, which differs slightly from the convolution
+arithmetic guide's. Here's a simple conversion table for the two:
+
++------------------+-------------------------------------------------------------------+
+| Theano           | Convolution arithmetic                                            |
++==================+===================================================================+
+| ``filters``      | 4D collection of kernels                                          |
++------------------+-------------------------------------------------------------------+
+| ``input_shape``  | (batch size, input channels, :math:`i_1`, :math:`i_2`)            |
++------------------+-------------------------------------------------------------------+
+| ``filter_shape`` | (output channels, input channels, :math:`k_1`, :math:`k_2`)       |
++------------------+-------------------------------------------------------------------+
+| ``border_mode``  | ``'valid'``, ``'half'``, ``'full'`` or (:math:`p_1`, :math:`p_2`) |
++------------------+-------------------------------------------------------------------+
+| ``subsample``    | (:math:`s_1`, :math:`s_2`)                                        |
++------------------+-------------------------------------------------------------------+
+
+For instance, the convolution shown above would correspond to the following
+Theano call:
+
+.. code-block:: python
+
+    output = theano.tensor.nnet.conv2d(
+        input, filters, input_shape=(5, 5), filter_shape=(1, 1, 3, 3),
+        border_mode=(1, 1), subsample=(2, 2))
+
+Convolution arithmetic
+======================
+
+No zero padding, unit strides
+-----------------------------
+
+The simplest case to analyze is when the kernel just slides across every
+position of the input (i.e., :math:`s = 1` and :math:`p = 0`).
+Here is an example for :math:`i = 4` and :math:`k = 3`:
+
+.. figure:: conv_arithmetic_figures/no_padding_no_strides.gif
+
+One way of defining the output size in this case is by the number of possible
+placements of the kernel on the input. Let's consider the width axis: the kernel
+starts on the leftmost part of the input feature map and slides by steps of one
+until it touches the right side of the input. The size of the output will be
+equal to the number of steps made, plus one, accounting for the initial position
+of the kernel. The same logic applies for the height axis.
+
+More formally, the following relationship can be inferred:
+
+.. admonition:: Relationship 1
+
+    .. code-block:: python
+
+        output = theano.tensor.nnet.conv2d(
+            input, filters, input_shape=(i, i), filter_shape=(c1, c2, k, k),
```

Comment:
Input shape should be 4D and to be consistent with the terminology table I'd use i1, i2